versioned/tiered-tests/src/main/java/com/dremio/nessie/versioned/tests/AbstractTestStore.java

```diff
@@ -365,10 +363,17 @@ public void deleteSelectedByConditionExpression1NoAlias() {
    */
   @Test
   public void deleteSelectedByConditionExpression2NoAlias() {
-    ConditionExpression conditionExpression2 = ConditionExpression.of(ExpressionFunction.equals(ExpressionFunction.size(COMMITS), TWO));
+    final ConditionExpression conditionExpression2 = ConditionExpression.of(ExpressionFunction.equals(ExpressionFunction.size(COMMITS), TWO));
     deleteConditional(SampleEntities.createBranch(random), ValueType.REF, true, Optional.of(conditionExpression2));
   }
 
+  private <T extends HasId> void putWithCondition(T sample, ValueType type, boolean putSucceeded) {
+    final ExpressionPath keyName = ExpressionPath.builder(Store.KEY_NAME).build();
+    final Entity id  = Entity.ofString(sample.getId().toString());
```

Comment:
This seems odd, why is the ID a String Entity and not a Binary one?

Reply:
updated to binary

Reply:
I'm unsure how/if this worked previously, which makes me wonder if the test is correct.